### ISSUE TITLE
ActiveType.cast should update STI column

### ISSUE THREAD

**Issue by foobear** [makandra/active_type#99]
Example:

```ruby
class Person::Base < ApplicationRecord
end

class Person::Alice < Person::Base
end

class Person::Bob < Person::Base
end
```

```ruby
>> ActiveType.cast(Person::Alice.new, Person::Bob).type
=> "Person::Alice"
```

I’d have expected the record’s `type` to change to `"Person::Bob"`.

This probably originates from `ActiveRecord#becomes`. There is `ActiveRecord#becomes!` but that also changes the _source_ record’s `type`:

```ruby
alice = Person::Alice.new
bob = alice.becomes!(Person::Bob)
>> alice.type
=> "Person::Bob"
>> bob.type
=> "Person::Bob"
```

I can’t see a reason why anyone would want that, so I believe `ActiveType.cast` should do this:

```ruby
alice = Person::Alice.new
bob = ActiveType.cast(alice, Person::Bob)
>> alice.type
=> "Person::Alice"
>> bob.type
=> "Person::Bob"
```

**Comment by kratob** [comment]
I agree this would be the expected behavior.

**Comment by denzelem** [comment]
Fix released as 0.7.4.

**Comment by kratob** [comment]
After the change, the source records `#type` changes as well.

Note that this is just a consequence of the fact that the source record and the casted record share *all* attributes, including `#type`.

We could discuss implementing `#cast` in a different way at the cost of performance.
